### PR TITLE
detect-replace.c:Convert unittests to FAIL/PASS API

### PR DESCRIPTION
**Pull request by Rai0Catodic0**
Ticket: [4054](https://redmine.openinfosecfoundation.org/issues/4054)

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [ :white_check_mark: ] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [ :white_check_mark: ] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR:  #8044 

Describe changes:
-  Modify detect-replace to use FAIL/PALL APIs
- Updated copyright year

### CODE REVIEW

File: src/detect-replace.c [hunk 1]
```diff
@@ -671,30 +663,20 @@ static int DetectReplaceParseTest03(void)
...
-    de_ctx->sig_list = SigInit(de_ctx,
-                               "alert tcp any any -> any any "
-                               "(msg:\"test\"; content:\"doh\"; replace:\"don\"; http_header; sid:238012;)");
-    if (de_ctx->sig_list != NULL) {
-        result = 0;
-        goto end;
-    }
+    de_ctx->sig_list = DetectEngineAppendSig(de_ctx,
+            "alert tcp any any -> any any "
+            "(msg:\"test\"; content:\"doh\"; replace:\"don\"; http_header; sid:238012;)");
```
Comment (jufajardini) [inline src/detect-replace.c:673]: I realize that I may have led to a mistake. I'm sorry about that.
We use `DetectEngineAppendSig()` when we have a signature declared.

You may follow the approach used in the test in `detect-content.c`, and just call it from within the maccro: https://github.com/OISF/suricata/blob/master/src/detect-content.c#L3084

File: src/detect-replace.c [hunk 2]
```diff
@@ -807,30 +753,18 @@ static int DetectReplaceParseTest07(void)
...
 
     de_ctx->flags |= DE_QUIET;
     de_ctx->sig_list = SigInit(de_ctx,
                                "alert tcp any any -> any any "
                                "(msg:\"test\"; content:\"don\"; replace:\"dou\"; content:\"jpg\"; http_header; sid:238012;)");
-    if (de_ctx->sig_list != NULL) {
-        result = 0;
-        goto end;
-    }
+    FAIL_IF_NOT_NULL(de_ctx->sig_list);
```
Comment (jufajardini) [inline src/detect-replace.c:763]: Please use the suggested approach of calling DetectEngineAppendSig from within the checking macro here as well

**Review by jufajardini — CHANGES_REQUESTED** [review]
Lots of improvements, thanks, and kudos!

Two things from my end (which I should have realized with the prior review, I apologize)
I left a comment about the usage of `DetectEngineAppendSig()`, as well, but offered an example.

We don't add the file extension, usually, in commit messages. So please do:
```
detect/replace: convert unittests to FAIL/PASS API

Tick #4054
```

After consulting a senior colleague, we've agreed that `DetectReplaceLongPatternMatchTest()` is also a good candidate for conversion and the clean up that you did in similar tests in this PR. :)

**Comment by Rai0Catodic0** [comment]
Closing in favour of a newer PR: #8049

**Pull request closed without merging**